include/common_package/common_node.hpp

```diff
@@ -68,6 +74,36 @@ namespace common_lib
             RCLCPP_DEBUG(this->get_logger(), "CommonNode::deactivate: Deactivated node");
         }
 
+        /**
+         * @brief Handles the completion of a job.
+         *
+         * This function sends a job_finished message with the given error code and payload.
+         * Additionally, deactivates the node.
+         *
+         * @param error_code The error code associated with the job completion (EXIT_SUCCESS == 0 if no error).
+         * @param payload The payload data associated with the job completion.
+         */
+        void job_finished(const uint8_t error_code, const nlohmann::json payload);
+
+        /**
+         * @brief Handles the completion of a job with the given error message. Can be used if your job fails and you just want to return an error message.
+         *
+         * This function formats the error message to JSON and sends a job_finished message with the error message.
+         * Additionally, deactivates the node.
+         * This function should only be used if your job failed.
+         *
+         * @param error_message The error message associated with the job completion.
+         */
+        void job_finished(const std::string error_message);
```

Comment:
Unnötiges kopieren vermeiden.

```suggestion
        void job_finished(const std::string &error_message);
```

Reply:
Ist gemacht.